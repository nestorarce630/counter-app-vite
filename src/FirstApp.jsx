import PropTypes from "prop-types";

/* const getResult = (a,b) =>{
    return a+b
}

const newMessage = {
    message: 'Hola Mundo',
    title: 'Fernando',

}; */

export const FirstApp = ( {title, subtitle,name} ) => {

  //console.log(props);


  return (
   <>
    <h1 data-testid="test-title"> {title} </h1>
    {/* <code>{getResult(1,2)}</code>  */}
    <p>{subtitle}</p>
    <p>{subtitle}</p>
    <p>{name}</p>
   </>
  )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
    name: 'Fernando Herrera',
    subtitle: 'No hay subtitulo',
    //title: 'No hay titulo',
}