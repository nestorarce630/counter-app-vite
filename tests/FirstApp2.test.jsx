/** @jest-environment jsdom */
import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp/>', () => {

    const title = 'Hola';
    const subtitle = 'Soy un subtitulo';
    test('debe de hacer match con el snapshot', () => { 
         
        const { container } = render (<firstApp title={title}/>);
        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar el mensaje "Hola"', () => {  
        render (<FirstApp title={title}/>);
        //console.log(screen.debug())
        expect(screen.getByText(title)).toBeTruthy();
    })

    test('debe de mostrar el titulo en un h1', () => {  
        render (<FirstApp title={title}/>);
        //console.log(screen.debug())
        expect(screen.getByRole('heading',{level: 1}).innerHTML).toContain(title);
    })

    test('debe de mostrar el subtitulo enviado por props', () => { 
        render(<FirstApp title={title} subtitle={subtitle}/>)
        expect(screen.getAllByText(subtitle).length).toBe(2)
    });
});