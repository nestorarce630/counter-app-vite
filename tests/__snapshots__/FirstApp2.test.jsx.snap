// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <firstapp
    title="Hola"
  />
</div>
`;
